api/v1beta3/authpolicy_types.go

```diff
@@ -293,47 +242,11 @@ func (ap *AuthPolicy) GetWrappedNamespace() gatewayapiv1.Namespace {
 	return gatewayapiv1.Namespace(ap.Namespace)
 }
 
-// GetRulesHostnames returns all hostnames referenced in the route selectors of the policy.
-func (ap *AuthPolicy) GetRulesHostnames() (ruleHosts []string) {
-	ruleHosts = make([]string, 0)
-
-	appendRuleHosts := func(obj RouteSelectorsGetter) {
-		for _, routeSelector := range obj.GetRouteSelectors() {
-			ruleHosts = append(ruleHosts, utils.HostnamesToStrings(routeSelector.Hostnames)...)
-		}
-	}
-
-	appendCommonSpecRuleHosts := func(c *AuthPolicyCommonSpec) {
-		if c.AuthScheme == nil {
-			return
-		}
-
-		for _, config := range c.AuthScheme.Authentication {
-			appendRuleHosts(config)
-		}
-		for _, config := range c.AuthScheme.Metadata {
-			appendRuleHosts(config)
-		}
-		for _, config := range c.AuthScheme.Authorization {
-			appendRuleHosts(config)
-		}
-		if response := c.AuthScheme.Response; response != nil {
-			for _, config := range response.Success.Headers {
-				appendRuleHosts(config)
-			}
-			for _, config := range response.Success.DynamicMetadata {
-				appendRuleHosts(config)
-			}
-		}
-		for _, config := range c.AuthScheme.Callbacks {
-			appendRuleHosts(config)
-		}
-	}
-
-	appendRuleHosts(ap.Spec.CommonSpec())
-	appendCommonSpecRuleHosts(ap.Spec.CommonSpec())
-
-	return
+// GetRulesHostnames
+// in v1beta2 this returned the list of route selectors
+// in v1beta3 this should work with section name, once implemented.
+func (ap *AuthPolicy) GetRulesHostnames() []string {
```

Comment:
Is there anything left that still depends on this? Maybe it's required to fully implement some interface by the actual method is no longer being used? (I didn't check.)